apps/web-connect/src/features/checkout/components/crossmint/EmbeddedCheckoutIFrame.tsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect, useState } from "react";
+
+import {
+    type CrossmintEmbeddedCheckoutProps,
+    type IncomingInternalEvent,
+    IncomingInternalEvents,
+    crossmintIFrameService,
+} from "@crossmint/client-sdk-base";
+
+type CrossmintEmbeddedCheckoutIFrameProps = CrossmintEmbeddedCheckoutProps & {
+    onInternalEvent?: (event: IncomingInternalEvent) => void;
+};
+
+export default function CrossmintEmbeddedCheckoutIFrame({
+    onInternalEvent,
+    ...props
+}: CrossmintEmbeddedCheckoutIFrameProps) {
+    const { getUrl, listenToEvents, listenToInternalEvents } = crossmintIFrameService(props);
+
+    const [height, setHeight] = useState(0);
+    const [url] = useState(getUrl(props));
+
+    // Public events
+    useEffect(() => {
+        const clearListener = listenToEvents((event) => props.onEvent?.(event.data));
+
+        return () => {
+            clearListener();
+        };
+    }, []);
```

Comment:
Same here, lint error just FYI:

`React Hook useEffect has missing dependencies: 'listenToEvents' and 'props'. Either include them or remove the dependency array. However, 'props' will change when *any* prop changes, so the preferred fix is to destructure the 'props' object outside of the useEffect call and refer to those specific props inside useEffect.`

Reply:
Save as above.